pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -3933,9 +3967,10 @@ def prepare_plots(self):
                     xunit = self.metadata["sweep_unit"]
                     xlabel = self.metadata["sweep_name"]
                 except KeyError:
-                    xlabel = self.raw_data_dict['sweep_parameter_names'][0]
-                    xunit = self.raw_data_dict['sweep_parameter_units'][0]
-
+                    xlabel = self.raw_data_dict['sweep_parameter_names']
+                    xunit = self.raw_data_dict['sweep_parameter_units']
+                if np.ndim(xunit) > 0:
+                    xunit = xunit[0]
```

Comment:
what about xlabel? Is it intended that xlabel might be a list?

Reply:
I am not entirely sure, so I added the same check for the xlabel as well :p